public/utils/savedObjectHelper.tsx

```diff
@@ -119,7 +119,11 @@ export const deleteAlertingAugmentVisSavedObj = async (
   monitorId: string
 ): Promise<void> => {
   const savedObjectLoader = getSavedAugmentVisLoader();
-  await savedObjectLoader.findAll().then(async (resp) => {
+  await savedObjectLoader.findAll('', 100, [], {
+      type: 'visualization',
+      id: visId,
+    }
+  ).then(async (resp) => {
     if (resp !== undefined) {
       const savedAugmentObjects = get(resp, 'hits', []);
       // gets all the saved object for this visualization
```

Comment:
You can remove the filtering below this since the raw response will already be filtered.